src/components/articles/SingleArticle.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { Row, Col } from 'antd';
+
+import AuthorHeadData from './singlearticle/AuthorArticleData';
+import './landingPage/Singlearticle.scss';
+
+const SingleArticle = ({
+  article, articleActions, history, editClick,
+}) => (
+  <div data-test="single-article">
+    <Row>
+      <Col span={18} offset={3}>
+        <Row>
+          <Col span={24} className="general-article-cols">
+            <AuthorHeadData
+              articleData={article}
+              articleActions={articleActions}
+              history={history}
+              editClick={editClick}
+            />
+          </Col>
+          <Col span={24} className="general-article-cols">
+            <p className="article-title">
+              {article.title}
+            </p>
+          </Col>
+          <Col span={24} className="general-article-cols">
+            <div className="article-description">
+              {article.description}
+            </div>
+          </Col>
+          {article.image ? (
+            <Col span={24} className="general-article-cols article-image">
+              <img src={article.image} alt={article.slug} />
+            </Col>
+          ) : null}
+
+          <Col span={24} className="general-article-cols">
+            <div
+              className="article-body"
+              dangerouslySetInnerHTML={{ __html: article.body }} // eslint-disable-line
+            />
+          </Col>
+          <Col span={24} className="tag-cont">
+            {article.tagList ? article.tagList.map(tag => (
```

Comment:
I may have not understood your question yesterday but I'm getting the data from react and returning a react element with the data inside it. That's why am using the map method here.
I have tried using foreach for this instance and it returns a blank space where the element should be displayed.
When I add a return statement in the foreach, then it works. However, array.map would be better for this instance rather than the foreach as I think it fits the context of usage based on your explanation and my research.